### ISSUE TITLE
New Rule: no-shell-dollars

### ISSUE THREAD

**Issue by BHyeonKim** [eslint/markdown#727]
### Rule details

Warns when a shell/terminal code block includes the $ prompt prefix on command lines. Copying such a snippet directly into a terminal (including the leading $) will fail, since $ is not part of the actual command. Code blocks that mix commands with their output are exempt, since the $ prefix in that case helps distinguish input lines from output lines rather than being copy-paste noise.

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
<!-- WARNING: `$` prompt included, breaks when copy-pasted into a terminal -->

$ npm install
$ npm run build


<!-- OK: no prompt prefix, safe to copy-paste directly -->

npm install
npm run build


<!-- OK: output is shown, so `$` helps distinguish commands from output and is not flagged -->

$ npm install
added 42 packages in 3s

$ npm run build
Build completed successfully
```

### Prior Art

markdownlint: https://github.com/DavidAnson/markdownlint/blob/main/doc/md014.md

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### AI acknowledgment

- [ ] I did not use AI to generate this issue report.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

Disclosure: I'm a participant of [open source contribution program OSSCA](https://github.com/eslint-ossca)

**Comment by DMartens** [comment]
I agree that we should add this rule because we accepted other rule proposals from existing rules in markdownlint.
As this is a new proposal, I would like other opinions from the team.

**Comment by lumirlumir** [comment]
> Disclosure: I'm a participant of [open source contribution program OSSCA](https://github.com/eslint-ossca): confirmed.

I am not sure whether this rule should be part of the core rules, as I don't think this is anti-pattern reporting but rather a user's preference. (https://eslint.org/docs/latest/use/core-concepts/glossary#stylistic-rule)

> I agree that we should add this rule because we have accepted other rule proposals based on existing rules in markdownlint.

Not all of them have been accepted. Looking at the closed issue history, stylistic choices have not been accepted even when they come from `markdownlint` or `remark-lint`.

[Another related issue in another ESLint Markdown plugin](https://github.com/ota-meshi/eslint-plugin-markdown-preferences/issues/201#issuecomment-3489423594) shows that it is rather a stylistic choice, so I am personally −1 on adding this to the core rules.

**Comment by uncoolclub** [comment]
Hi! I'm currently implementing this rule in `eslint-markdown` ([#674](https://github.com/lumirlumir/npm-eslint-markdown/pull/674)), so here is a short opinion from that experience.

I think this sits closer to a stylistic rule than to a problem. ESLint's glossary describes a stylistic rule as one that "enforces a preference rather than a logical issue", including "consistent choices between equivalent syntaxes". `$ npm install` and `npm install` are exactly that: the document renders the same and documents the same command either way. What the rule reports is friction when copying, not an incorrect document.

The exemption makes this clearer. The nvm README contains both forms, and the only difference is whether the author pasted the output:

```sh
$ nvm install 24
Now using node v24.14.0 (npm v11.9.0)
```

```sh
$ echo "5.9" > .nvmrc
```

The first is exempt, the second is reported, and both are in the same README with the same intent behind them. Prompt characters vary as well (`$`, `%`, `#`, `>`, `PS>`), so any implementation has to encode one house style.

Stylistic ports from `markdownlint` have been closed as not planned before (#682, #612, #689, #387, #448), and several of them now live in plugins instead. So if this rule is added here, I think it fits better as an opt-in rule, the way `no-html` and `no-bare-urls` sit outside `recommended`, rather than as one enabled by default.

**Comment by BHyeonKim** [comment]
I understand that this rule reflects a user preference rather than pointing out an anti-pattern.

As @lumirlumir mentioned, it probably shouldn't be included as a core rule. 
Since @uncoolclub has already submitted a PR for this rule to the eslint-markdown plugin, I'll close this issue.

Thanks!
